Assets/Scripts/Player/ScriptedJump.cs

```diff
@@ -0,0 +1,47 @@
+﻿using UnityEngine;
+
+[DefaultExecutionOrder(-1)]
+public class ScriptedJump : MonoBehaviour {
+	[SerializeField] private float jumpStartTime = 0.1f;
+	[SerializeField] private float jumpLength = 0.05f;
+	[SerializeField] private float moveLength = 0.45f;
+
+	private PlayerController playerController;
+	private SimpleTimer jumpStartTimer;
+	private SimpleTimer jumpLengthTimer;
+	private SimpleTimer moveTimer;
+
+	private bool firstFrame = true;
+
+	private void Awake() {
+		playerController = GetComponent<PlayerController>();
+	}
```

Comment:
Maybe disable controls fully here? Just to fully prevent the player from doing anything.

Reply:
I can't just set `PlayerController.AllowControls` as that would also prevent my fake inputs from doing anything. An option could be to always overwrite all the inputs from my script while its running. But do you really think its necessary to do something like this? Its just used for a very short duration and I don't think the player doing anything really does any harm.

Reply:
ahh right, Perhaps controlling input then is not the correct way but rather call functions to control the player. Could leave this as is for now since the other way would require some refactoring of the playercontroller. I'm primarily thinking about dash, airjump and transform so a bad way could be to just disable those.